mypy/nodes.py

```diff
@@ -1946,6 +1946,27 @@ def __init__(self, names: 'SymbolTable', defn: ClassDef, module_name: str) -> No
             for vd in defn.type_vars:
                 self.type_vars.append(vd.name)
 
+    declared_metaclass = None  # type: Optional[mypy.types.Instance]
+    metaclass_type = None  # type: Optional[mypy.types.Instance]
+
+    def calculate_metaclass_type(self) -> 'Optional[mypy.types.Instance]':
+        if self.mro is None:
+            # XXX why does this happen?
```

Comment:
I think I will remove the call from deserialize, since it shouldn't be done yet - it should be done from fixup.py. Does it sound reasonable?